### PR TITLE
Bump version number and ensure requirements.txt are installed properly on pip install

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Addresses #242. This PR may require more work after it is merged in because we can only see the final results after a merge into master.

Now that Omer has completely integrated deepcell with ark-analysis, we need to bump the version number. However, another issue needs to be addressed: running `pip install ark-analysis` will only install `ark-analysis`, but not any of the other dependencies.

**How did you implement your changes**

We use `deepcell-toolbox` as a guiding point for our changes.

**Remaining issues**

It seems like `deepcell-toolbox` specifically implemented some of their changes because they need to install `deepcell` as well. I'm not sure how much is actually applicable to us, it may potentially not work. If so, we'll need to open another PR to address the issues that come up.

### CODE REVIEW

**Comment by alex-l-kong** [comment]
So I literally just copied most everything that Will did on `deepcell-toolbox`, which I confirmed does install dependencies correctly. If this still doesn't fix our `pip install` issue with dependencies, then I think we need to ask Will for some extra help.

**Comment by ngreenwald** [comment]
Did you take this from `deeepcell-tf` or `deepcell-toolbox`? You mentioned this could be due to a requirements issue, but the toolbox doesn't have `deepcell-tf` as a requirement

**Comment by alex-l-kong** [comment]
> Did you take this from `deeepcell-tf` or `deepcell-toolbox`? You mentioned this could be due to a requirements issue, but the toolbox doesn't have `deepcell-tf` as a requirement

I took this from `deepcell-toolbox`. But now that you mention it, perhaps I should've been looking at `kiosk-client`. Without any complicated `setup.py` jazz, `pip` installed it fine: the requirements all downloaded properly. I am curious about the extra Docker commands in `.travis.yml` are doing though. Wonder if they help install the requirements in `requirements.txt`.

**Comment by ngreenwald** [comment]
Maybe ask Will? If we can keep things simpler that would be better

**Comment by alex-l-kong** [comment]
> Maybe ask Will? If we can keep things simpler that would be better

Ah OK, now I know why I used `deepcell-toolbox` as a reference. Will told me that its `setup.py` was configured properly, as well as the `.travis.yml` file which explicitly ran it. I decided to copy the `BuildExtension` class as well because `deepcell-toolbox` does that in their `setup.py` as well.

I'll ask Will what that function does, and if he says its purely for building `deepcell` on their end, then I think we can safely remove it and hope that we only need to run `setup.py` in Travis for it to properly build the extensions we need.

**Comment by alex-l-kong** [comment]
Just heard back from Will. He said that the `BuildExtension` class was only if you needed to compile C files for `cython`. This is required for `deepcell-toolbox` because they actually need this, but we don't. So we're in the clear to remove it.

The key is to have the `travis_retry python setup.py` in `.travis.yml` so Travis actually runs `setup.py` (which includes the dependencies) on their end before deploying to PyPI.

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Sounds good, let me know once it's ready

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
+import setuptools
 from os import path
+from distutils.command.build_ext import build_ext as DistUtilsBuildExt
 from setuptools import setup, find_packages
 
-VERSION = '0.2.6'
+VERSION = '0.3.6'
```
Comment (ngreenwald) [inline setup.py:6]: If we're bumping this, it should 0.2.7 right?

**Review by ngreenwald — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (alex-l-kong) [reply setup.py:6]: Ah got it, will fix.

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**